src/Bicep.LangServer/BicepParamsCompilationManager.cs

```diff
@@ -35,8 +48,17 @@ public void RefreshCompilation(DocumentUri uri, bool reloadBicepConfig = false)
 
         public void UpsertCompilation(DocumentUri uri, int? version, string text, string? languageId = null)
         {
-            var semanticModel = new ParamsSemanticModel(SourceFileFactory.CreateBicepParamFile(uri.ToUri(), text));
+            var paramsFile = SourceFileFactory.CreateBicepParamFile(uri.ToUri(), text);
+            var semanticModel = new ParamsSemanticModel(paramsFile);
+            var usingDeclarations = paramsFile.ProgramSyntax.Children.OfType<UsingDeclarationSyntax>();
 
+            if (PathHelper.TryGetUsingPath(usingDeclarations.SingleOrDefault(), out var bicepFilePath, out var _) && Uri.TryCreate(uri.ToUri(), bicepFilePath, out var bicepFileUri))
+            {
+                var bicepConfig = bicepConfigurationManager.GetConfiguration(bicepFileUri);
+                var bicepCompilationContext = bicepCompilationContextProvider.Create(new Workspace(), bicepFileUri, new Dictionary<ISourceFile, ISemanticModel>().ToImmutableDictionary(), bicepConfig, new LinterAnalyzer(bicepConfig));
+                semanticModel = new ParamsSemanticModel(paramsFile, bicepCompilationContext.Compilation);
```

Comment:
Since we removed setter for the ```Compilation``` in ```ParamsSemanticModel```, @jeskew do you think reconstructing the ```ParamsSemanticModel``` is okay? (if bicep file path can be prased)

Reply:
I think it's fine for now. There are a few options if you want to avoid calling the constructor twice, including but not limited to:

1. Doing the `using` lookup and perform the compilation (if necessary) in the constructor. You could pass a compiler function (with a signature like `(Uri) => Compilation`) as a constructor argument if this should be handled differently in the language server, and the compilation itself could be performed lazily if that seems like too much computation to do upfront.
2. Adding a `WithCompilation(Compilation c)` method to the `ParamsSemanticModel` class that returns a new model. This would let you reuse anything calculated during the prior constructor invocation while still keeping the class immutable.